src/java/test/org/apache/zookeeper/client/ZKClientConfigTest.java

```diff
@@ -0,0 +1,141 @@
+import static org.apache.zookeeper.client.ZKClientConfig.LOGIN_CONTEXT_NAME_KEY;
+import static org.apache.zookeeper.client.ZKClientConfig.SECURE_CLIENT;
+import static org.apache.zookeeper.client.ZKClientConfig.ZK_SASL_CLIENT_USERNAME;
+import static org.apache.zookeeper.client.ZKClientConfig.ZOOKEEPER_CLIENT_CNXN_SOCKET;
+import static org.apache.zookeeper.client.ZKClientConfig.ZOOKEEPER_SERVER_REALM;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+import org.apache.zookeeper.server.quorum.QuorumPeerConfig.ConfigException;
+import org.junit.Test;
+
+public class ZKClientConfigTest {
+
+    @Test
+    public void testDefaultConfiguration() {
+
+        String[] values = new String[] { "zookeeper1", "Client1", "true", "zookeeper/hadoop.hadoop.com", "true",
+                "ClientCnxnSocketNetty", "true" };
+        String[] properties = new String[] { ZK_SASL_CLIENT_USERNAME, LOGIN_CONTEXT_NAME_KEY, ENABLE_CLIENT_SASL_KEY,
+                ZOOKEEPER_SERVER_REALM, DISABLE_AUTO_WATCH_RESET, ZOOKEEPER_CLIENT_CNXN_SOCKET, SECURE_CLIENT };
+        assertEquals(properties.length,values.length);
+        for (int i = 0; i < properties.length; i++) {
+            String prop = properties[i];
+            System.setProperty(prop, values[i]);
+        }
+        /**
+         * ZKClientConfig should get initialized with system properties
+         */
+        ZKClientConfig conf = new ZKClientConfig();
+        for (int i = 0; i < properties.length; i++) {
+            String prop = properties[i];
+            String result = conf.getProperty(prop);
+            assertEquals(values[i], result);
+        }
+
+        /**
+         * clear properties
+         */
+        for (String prop : properties) {
+            System.clearProperty(prop);
+        }
+
+        conf = new ZKClientConfig();
+        /**
+         * test that all the properties are null
+         */
+        for (String prop : properties) {
+            String result = conf.getProperty(prop);
+            assertNull(result);
+        }
+
+    }
+
+    @Test
+    public void testSystemProprtyValue() {
+        String clientName = "zookeeper1";
+        System.setProperty(ZK_SASL_CLIENT_USERNAME, clientName);
+
+        ZKClientConfig conf = new ZKClientConfig();
+        assertEquals(conf.getProperty(ZK_SASL_CLIENT_USERNAME), clientName);
+
+        String newClientName = "zookeeper2";
+        conf.setProperty(ZK_SASL_CLIENT_USERNAME, newClientName);
+
+        assertEquals(conf.getProperty(ZK_SASL_CLIENT_USERNAME), newClientName);
+    }
+
+    @Test
+    public void testReadConfigurationFile() throws IOException, ConfigException {
+        File file = new File("test.conf");
+        OutputStream io = new FileOutputStream(file);
```

Comment:
Why not use a ``PrintWriter`` here? The code from line 100 to 107 would be:
```
PrintWriter io = new PrintWriter(file);
io.println(ENABLE_CLIENT_SASL_KEY + "=true");
io.println(ZK_SASL_CLIENT_USERNAME + "=ZK");
io.println(LOGIN_CONTEXT_NAME_KEY + "=MyClient");
io.println(ZOOKEEPER_SERVER_REALM + "=HADOOP.COM");
io.println("dummyProperty=dummyValue");
io.flush();
```

Or yet

```
        Properties props = new Properties();
        props.setProperty(ENABLE_CLIENT_SASL_KEY, "true");
        props.setProperty(ZK_SASL_CLIENT_USERNAME, "ZK");
        props.setProperty(LOGIN_CONTEXT_NAME_KEY, "MyClient");
        props.setProperty(ZOOKEEPER_SERVER_REALM, "HADOOP.COM");
        props.setProperty("dummyProperty", "dummyValue");
        props.store(file, null);
        io.flush();
```